test/snoopi_deep.jl

```diff
@@ -944,6 +945,163 @@ end
+
+    prof2 = SnoopCompileCore.finish_snoopi_deep(t2)
+
+    @test Set(_name.(SnoopCompile.flatten(prof1))) == Set([:ROOT, :foo1, :foo2])
+    @test Set(_name.(SnoopCompile.flatten(prof2))) == Set([:ROOT, :foo2, :foo3])
+
+    # Test Cleanup
+    @test isempty(SnoopCompileCore.SnoopiDeepParallelism.invocations)
+    @test isempty(Core.Compiler.Timings._timings[1].children)
+end
+
+@testset "reentrant concurrent profiles 3 - nested" begin
+    # Warmup
+    @eval foo1(x) = x+2
+    @eval foo1(2)
+
+    # Test:
+    local prof1, prof2, prof3
+    prof1 = SnoopCompileCore.@snoopi_deep begin
+        @eval foo1(x) = x+2
+        @eval foo1(2)
+        prof2 = SnoopCompileCore.@snoopi_deep begin
+            @eval foo2(x) = x+2
+            @eval foo2(2)
+            prof3 = SnoopCompileCore.@snoopi_deep begin
+                @eval foo3(x) = x+2
+                @eval foo3(2)
+            end
+            @eval foo4(x) = x+2
+            @eval foo4(2)
+        end
+        @eval foo5(x) = x+2
+        @eval foo5(2)
+    end
+
+    @test Set(_name.(SnoopCompile.flatten(prof1))) == Set([:ROOT, :foo1, :foo2, :foo3, :foo4, :foo5])
+    @test Set(_name.(SnoopCompile.flatten(prof2))) == Set([:ROOT,        :foo2, :foo3, :foo4])
+    @test Set(_name.(SnoopCompile.flatten(prof3))) == Set([:ROOT,               :foo3])
+
+    # Test Cleanup
+    @test isempty(SnoopCompileCore.SnoopiDeepParallelism.invocations)
+    @test isempty(Core.Compiler.Timings._timings[1].children)
+end
+
+@testset "reentrant concurrent profiles 3 - parallelism + accurate timing" begin
+    # Warmup
+    @eval foo1(x) = x+2
+    @eval foo1(2)
+
+    # Test:
+    local ts
+    snoop_times = Float64[0.0, 0.0, 0.0, 0.0]
+    # Run it twice to ensure we warmup the eval block
+    for _ in 1:2
+        @sync begin
+            ts = [
+                Threads.@spawn begin
+                    sleep((i-1) / 10)  # (Divide by 10 so the test isn't too slow)
+                    snoop_time = @timed SnoopCompile.@snoopi_deep @eval begin
+                        $(Symbol("foo$i"))(x) = x + 1
+                        sleep(1.5 / 10)
+                        $(Symbol("foo$i"))(2)
+                    end
+                    snoop_times[i] = snoop_time.time
+                    return snoop_time.value
+                end
+                for i in 1:4
+            ]
+        end
+    end
+    profs = fetch.(ts)
+
+    @test Set(_name.(SnoopCompile.flatten(profs[1]))) == Set([:ROOT, :foo1])
+    @test Set(_name.(SnoopCompile.flatten(profs[2]))) == Set([:ROOT, :foo1, :foo2])
+    @test Set(_name.(SnoopCompile.flatten(profs[3]))) == Set([:ROOT,        :foo2, :foo3])
+    @test Set(_name.(SnoopCompile.flatten(profs[4]))) == Set([:ROOT,               :foo3, :foo4])
+
+    # Test the sanity of the reported Timings
+    @testset for i in eachindex(profs)
+        prof = profs[i]
+        # Test that the time for the inference is accounted for
+        @test 0.15 < prof.mi_timing.exclusive_time
```

Comment:
I fear the definitions of these times has already become hazy to me 😅. This one? 🤔How do know that the exclusive rather than inclusive time should be over 0.15 seconds?